functions/src/notifications/deliverNotifications.ts

```diff
@@ -128,45 +81,155 @@ export const deliverNotifications = functions.pubsub
+    await userDoc.ref.update({ nextDigestAt })
+
+    console.log(`Updated nextDigestAt for ${userDoc.id} to ${nextDigestAt}`)
+  })
+
+  // Wait for all email documents to be created
+  await Promise.all(emailPromises)
+}
+
+// TODO: Unit tests
+const buildDigestData = async (user: User, userId: string, now: Timestamp) => {
+  const startDate = getNotificationStartDate(user.notificationFrequency, now)
+
+  const notificationsSnapshot = await db
+    .collection(`users/${userId}/userNotificationFeed`)
+    .where("notification.type", "==", "testimony") // Email digest only cares about testimony
+    .where("notification.timestamp", ">=", startDate)
+    .where("notification.timestamp", "<", now)
+    .get()
+
+  const billsById: { [billId: string]: BillDigest } = {}
+  const usersById: { [userId: string]: UserDigest } = {}
+
+  notificationsSnapshot.docs.forEach(notificationDoc => {
+    const { notification } =
+      notificationDoc.data() as TestimonySubmissionNotificationFields
+
+    if (notification.isBillMatch) {
+      if (billsById[notification.billId]) {
+        const bill = billsById[notification.billId]
+
+        switch (notification.position) {
+          case "endorse":
+            bill.endorseCount++
+            break
+          case "neutral":
+            bill.neutralCount++
+            break
+          case "oppose":
+            bill.opposeCount++
+            break
+          default:
+            console.error(`Unknown position: ${notification.position}`)
+            break
+        }
+      } else {
+        billsById[notification.billId] = {
+          billId: notification.billId,
+          billName: notification.header,
+          billCourt: notification.court,
+          endorseCount: notification.position === "endorse" ? 1 : 0,
+          neutralCount: notification.position === "neutral" ? 1 : 0,
+          opposeCount: notification.position === "oppose" ? 1 : 0
+        }
       }
+    }
 
-      await doc.ref.update({ nextDigestAt })
-    })
+    if (notification.isUserMatch) {
+      const billResult = {
+        billId: notification.billId,
+        court: notification.court,
+        position: notification.position as Position
+      }
+      if (usersById[notification.authorUid]) {
+        const user = usersById[notification.authorUid]
+        user.bills.push(billResult)
+        user.newTestimonyCount++
+      } else {
+        usersById[notification.authorUid] = {
+          userId: notification.authorUid,
+          userName: notification.subheader,
+          bills: [billResult],
+          newTestimonyCount: 1
+        }
+      }
+    }
+  })
 
-    // Wait for all email documents to be created
-    await Promise.all(emailPromises)
+  const bills = Object.values(billsById).sort((a, b) => {
+    return (
+      b.endorseCount +
+      b.neutralCount +
+      b.opposeCount -
+      a.endorseCount -
+      a.neutralCount -
+      a.opposeCount
```

Comment:
Are we sorting this by total number of votes? If so, maybe we change it to ` (b.endorseCount +
      b.neutralCount +
      b.opposeCount) -
      (a.endorseCount +
      a.neutralCount +
      a.opposeCount)` Just to make it a little bit more clear?